sdk/core/azure-core/azure/core/paging.py

```diff
@@ -24,65 +24,360 @@
+        :type client: ~azure.core.PipelineClient
+        :return: Next request for the pager to make
+        :rtype: ~azure.core.pipeline.transport.HttpRequest
+        """
+        raise NotImplementedError("This method needs to be implemented")
+
+    # extracting data from response
+
+    def get_list_elements(self, pipeline_response, deserialized, item_name="value"):
+        # type: (HttpResponse, ResponseType, str) -> Iterable[ReturnType]
+        """Extract the list elements from the current page to return to users
+
+        :param pipeline_response: The immediate response returned from the pipeline
+        :type pipeline_response: ~azure.core.pipeline.transport.HttpResponse
+        :param object deserialized: The deserialized pipeline_response
+        :param optional[str] item_name: The property name on the response that houses
+         the list elements. Default is `value`.
+        :return: The iterable of items extracted out of paging response
+        :rtype: iterable[object]
+        """
+        raise NotImplementedError("This method needs to be implemented")
+
+    def mutate_list(self, pipeline_response, list_of_elem, cls=None):
+        # type: (HttpResponse, Iterable[ReturnType], Optional[Callable]) -> Iterable[ReturnType]
+        """Mutate list of elements in current page, i.e. if users passed in a cls calback
+
+        :param pipeline_response: The immediate response returned from the pipeline
+        :type pipeline_response: ~azure.core.pipeline.transport.HttpResponse
+        :param list[object] list_of_elem: The iterable of items we will return to users
+         in a page.
+        :param optional[callable] cls: Callback to mutate the list.
+        :return: The final list of iterable items we will return to users
+        :rtype: iterable[object]
+        """
+        raise NotImplementedError("This method needs to be implemented")
+
+    def get_continuation_token(self, pipeline_response, deserialized, continuation_token_location=None):
+        # type: (HttpResponse, ResponseType, Optional[str]) -> Any
+        """Get the continuation token from the current page. This operation returning None signals the end of paging.
+        Continuation token can be mutated here as well.
+
+        :param pipeline_response: The immediate response returned from the pipeline
+        :type pipeline_response: ~azure.core.pipeline.transport.HttpResponse
+        :param object deserialized: The deserialized pipeline_response
+        :param optional[str] continuation_token_location: Property name on the response object that houses the
+         continuation token. Defaults to `None`, as `None` is an acceptable value for a continuation token location.
+         It means that there's no continuation token on the response object.
+        :return: The continuation token
+        :rtype: any
+        """
+
+        raise NotImplementedError("This method needs to be implemented")
+
+class NextLinkPagingMethod(PagingMethodABC):
```

Comment:
Sadly couldn't keep `CallbackPagingMethod` as the base, since `NextLinkPagingMethod` requires the `client` that is passed to us through the `get_next_request` method signature. Made `NextLlinkPagingMetjhod` the base bc it has no required parameters